site/docs/v1/tech/guides/user-actions.adoc

```diff
@@ -0,0 +1,906 @@
+
+=== Create PiedPiper API to Listen for Expiry and Call Prevent Login Action
+
+The final piece of code you'll add to `app.js` is a little more complex. The `expire` route below is called by FusionAuth when the user's subscription Action instance ends. To ban the user from logging in after this time, PiedPiper applies the [field]#Prevent Login# Action to the user by calling the FusionAuth API.
+
+[source,js]
+----
+app.post('/expire', async function(req, res) {
+  console.log('Incoming Request to PiedPiper Expiry:');
+  console.log(req.body);
+  console.log('');
+  if (req.body.event.action === 'Subscribe' && req.body.event.phase === 'end') {
+    try {
+      const request = {
+        action: {
+          actioneeUserId: req.body.event.actioneeUserId,
+          actionerUserId: req.body.event.actionerUserId,
+          applicationIds: ['e9fdb985-9173-4e01-9d73-ac2d60d1dc8e'],
+          emailUser: false,
+          expiry: 8223372036854775806, // the end of time
+          notifyUser: false,
+          reasonId: '28b0dd40-3a65-48ae-8eb3-4d63d253180a', // subscription expired reason
+          userActionId: 'b96a0548-e87c-42dd-887c-31294ca10c8b' //ban action
+        },
+        broadcast: false
+      };
+      const fusion = new client.FusionAuthClient('FTQkSoanK7ObbNjOoU69WDVclfTx8L_zfEJbdR8M0xu-jKotV0iQZiQh', 'http://localhost:9011');
+      const clientResponse = await fusion.actionUser(request);
+      if (!clientResponse.wasSuccessful)
+        throw Error(clientResponse);
+      console.info('User banned successfully');
+    }
+    catch (e) {
+      console.error('Error handling expiry: ');
+      console.dir(e, { depth: null });
+    }
+  }
+  res.sendStatus(200);
+});
+----
+
+== Testing
+
+In this last section, you'll see how Actions work by applying them and watching the emails and webhooks be triggered.
+
+=== Start PiedPiper
+
+Run the PiedPiper Node.js app by typing the following in a terminal.
+
+[source,bash]
+----
+npm run start
+----
+
+=== Apply Subscription Action
+
+Let's start testing by applying the subscription Action to the user. In reality, your app would do this in code once the user has paid, but for now we'll do it in a new terminal.
+
+[NOTE]
+====
+You'll need to install `curl` if it's not already installed.
+====
+
+In the following code, you will replace the values of [filed]#actioneeUserId# and [field]#actionerUserId# with the values you recorded earlier for the reader and administrator users respectively.
+
+To test out the workflow, you can let the subscription expire after 60 seconds. From the https://fusionauth.io/dev-tools/date-time[FusionAuth Date-Time tool], copy the [field]#Milliseconds# value, add `60000` (60 seconds) to it, and paste it into the `"expiry"` field below. This will ensure the subscription action expires immediately. If you're on Linux, use the Option 2 code snippet to set the [field]#expiry# value automatically.
```

Comment:
```suggestion
To test out the workflow, you can let the subscription expire after 60 seconds. From the https://fusionauth.io/dev-tools/date-time[FusionAuth Date-Time tool], copy the [field]#Milliseconds# value, add `60000` (60 seconds) to it, and paste it into the `"expiry"` field below. This will ensure the subscription action expires quickly. If you're on Linux, use the Option 2 code snippet to set the [field]#expiry# value automatically.
```